pkg/controller/apimanagerbackup/apimanagerbackup_logic_reconciler.go

```diff
@@ -0,0 +1,398 @@
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		substepFinished := true
+		if existing.Status.Succeeded != *desired.Spec.Completions {
+			r.Logger().Info("Job has still not finished", "Job Name", desired.Name, "Actively running Pods", existing.Status.Active, "Failed pods", existing.Status.Failed)
+			return reconcile.Result{Requeue: true}, nil
+		}
+
+		r.cr.Status.APIManagerCustomResourceBackupSubStepFinished = &substepFinished
+		err = r.UpdateResourceStatus(r.cr)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		r.Logger().Info("Job finished successfully. Requeing", "Job Name", desired.Name)
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Check if cleanup substep has completed
+	if !r.cr.APIManagerCustomResourceCleanupSubStepFinished() {
+		common.TagToObjectDeleteWithPropagationPolicy(desired, metav1.DeletePropagationForeground)
+		err := r.ReconcileResource(existing, desired, reconcilers.CreateOnlyMutator)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		err = r.GetResource(common.ObjectKey(desired), existing)
+		if err != nil && !errors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+
+		substepFinished := true
+		r.cr.Status.APIManagerCustomResourceCleanupSubStepFinished = &substepFinished
+		err = r.UpdateResourceStatus(r.cr)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		r.Logger().Info("Job cleant up successfully. Requeuing", "Job Name", desired.Name)
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *APIManagerBackupLogicReconciler) reconcileBackupSystemFileStoragePVCToPVCJob() (reconcile.Result, error) {
+	desired := r.apiManagerBackup.BackupSystemFileStoragePVCToPVCJob()
+	if desired == nil {
+		return reconcile.Result{}, nil
+	}
+
+	// TODO create Mutator function for Jobs??
+	existing := &batchv1.Job{}
+	if r.cr.SystemFileStorageBackupStepFinished() {
+		return reconcile.Result{}, nil
+	}
+
+	// Check if backup substep has completed
+	if !r.cr.SystemFileStorageBackupSubStepFinished() {
+		err := r.ReconcileResource(existing, desired, reconcilers.CreateOnlyMutator)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		backupFinished := true
+		if existing.Status.Succeeded != *desired.Spec.Completions {
+			r.Logger().Info("Job has still not finished", "Job Name", desired.Name, "Actively running Pods", existing.Status.Active, "Failed pods", existing.Status.Failed)
+			return reconcile.Result{Requeue: true}, nil
+		}
+
+		r.cr.Status.SystemFileStorageBackupSubStepFinished = &backupFinished
+		err = r.UpdateResourceStatus(r.cr)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		r.Logger().Info("Job finished successfully. Requeing", "Job Name", desired.Name)
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Check if cleanup substep has completed
+	if !r.cr.SystemFileStorageCleanupSubStepFinished() {
+		// TODO delete performed with ReconcileResource does not have the option to delete
```

Comment:
In Kubernetes >=1.9, for all kinds in the apps/v1 group version, dependent objects are deleted by default. Why do you need to specify cascading deletion policy?

Reply:
For what I observed, when using Go client they don't by default. When using kubectl it's the default behaviour but it isn't when using Go apparently

Reply:
Marking as resolved. If you have any comment about this please tell me.